transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+# limitations under the License.
+################################################################################
+
+import io
+import zipfile
+import time
+from argparse import ArgumentParser, Namespace
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import filetype
+import pyarrow as pa
+from data_processing.transform import AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import TransformUtils, get_logger, str2bool
+from docling.datamodel.base_models import DocumentStream
+from docling.datamodel.document import ConvertedDocument, DocumentConversionInput
+from docling.document_converter import DocumentConverter
+from docling.pipeline.standard_model_pipeline import PipelineOptions
+
+
+logger = get_logger(__name__)
+
+shortname = "pdf2md"
+cli_prefix = f"{shortname}_"
+pdf2md_modelsdir_key = f"{shortname}_modelsdir"
+pdf2md_download_models_key = f"{shortname}_download_models"
+pdf2md_do_table_structure_key = f"{shortname}_do_table_structure"
+pdf2md_do_ocr_key = f"{shortname}_do_ocr"
+
+
+class PdfToMdTransform(AbstractBinaryTransform):
+    """ """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, LangSelectorTransformRuntime.  If running inside the RayMutatingDriver,
+        these will be provided by that class with help from the RayMutatingDriver.
+        """
+
+        super().__init__(config)
+        
+        self.models_dir = Path(config.get(pdf2md_modelsdir_key, "scratch"))
+        self.do_table_structure = config.get(pdf2md_do_table_structure_key, True)
+        self.do_ocr = config.get(pdf2md_do_ocr_key, False)
+
+        download_models = config.get(pdf2md_download_models_key, False)
+        if download_models:
+            print("Downloading models")
+            logger.info("Downloading models")
+            DocumentConverter.download_models_hf(local_dir=self.models_dir)
+        print("Initializing models")
+        logger.info("Initializing models")
+        pipeline_options = PipelineOptions(
+            do_table_structure=self.do_table_structure,
+            do_ocr=self.do_ocr,
+        )
+        self._converter = DocumentConverter(artifacts_path=self.models_dir, pipeline_options=pipeline_options)
+
+        # self.doc_counter = Counter("worker_pdf_doc_count", "Number of PDF documents converted by the worker")
+        # self.page_counter = Counter("worker_pdf_pages_count", "Number of PDF pages converted by the worker")
+        # self.page_convert_gauge = Gauge("worker_pdf_page_avg_convert_time", "Average time for converting a single PDF page on each worker")
+        # self.doc_convert_gauge = Gauge("worker_pdf_convert_time", "Time spent converting a single document")
+
+    def _update_metrics(self, num_pages: int, elapse_time: float):
```

Comment:
why is this here?

Reply:
this is overloaded in the ray implementation

Reply:
I get it, but it does not belong here